graders/python/python_autograder/code_feedback.py

```diff
@@ -334,6 +334,81 @@ def bad(msg: str) -> Literal[False]:
 
         return True
 
+    @classmethod
+    def check_dict(
+        cls,
+        name: str,
+        ref: dict[Any, Any],
+        data: dict[Any, Any],
+        *,
+        only_keys: None | list[str] = None,
```

Comment:
Maybe `target_keys` is a better variable name? The current name makes it seem like this should be a boolean. I also think this should be a set since we don't want duplicates.

Reply:
Fixed in b0154849a9187cec010fdefb137a06043532db88.

Reply:
We would have to do `list -> set -> sort -> list` to support any sort of deterministic ordering on the keys.

Reply:
In [fa3fde1](https://github.com/PrairieLearn/PrairieLearn/pull/9436/commits/fa3fde1fd0dd6ff3c46b47d15e195e4edd3984b1) I support a set. No code changes needed.